src/cgroups/cgroup_utils.c

```diff
@@ -770,6 +770,8 @@ int cgroup_walkup_to_root(int cgroup2_root_fd, int hierarchy_fd,
 	val = readat_file(dir_fd, file);
 	if (is_empty_string(val) || strcmp(val, "max") == 0) {
 		no_limit = true;
+		if (same_file(cgroup2_root_fd, dir_fd))
```

Comment:
doesn't this now potentially see `-EBADF` as `cgroup2_root_fd` and thus in the `fstat()` call in `same_file()`?

Reply:
fixed :)